api/librdb-api.h

```diff
@@ -207,11 +206,13 @@ typedef struct RdbHandlersDataCallbacks {
     RdbRes (*handleHashElement)(RdbParser *p, void *userData, RdbBulk field, RdbBulk elm, uint64_t totalNumElm);
     RdbRes (*handleSetElement)(RdbParser *p, void *userData, RdbBulk elm, uint64_t totalNumElm);
     RdbRes (*handleZsetElement)(RdbParser *p, void *userData, RdbBulk elm, double score, uint64_t totalNumElm);
-    /*** TODO: RdbHandlersDataCallbacks: handleStreamNewGroup, handleStreamMetadata, handleStreamElement, handleStreamConsumerPendingEntry ***/
-    RdbRes (*handleStreamNewGroup)(RdbParser *p, void *userData, RdbBulk grpName, RdbStreamID lastId);
+    /*** TODO: RdbHandlersDataCallbacks: stream stuff ... ***/
     RdbRes (*handleStreamMetadata)(RdbParser *p, void *userData, RdbStreamMeta *meta);
     RdbRes (*handleStreamElement)(RdbParser *p, void *userData, RdbStreamID *id, RdbBulk field, RdbBulk value, uint64_t totalNumElm);
-    RdbRes (*handleStreamConsumerPendingEntry)(RdbParser *p, void *userData, RdbBulk groupName, RdbStreamPendingEntry *pendingEntry);
+    RdbRes (*handleStreamNewCGroup)(RdbParser *p, void *userData, RdbBulk grpName, RdbStreamID lastId);
+    RdbRes (*handleStreamCGroupPendingEntry)(RdbParser *p, void *userData, RdbBulk grpName, RdbBulk consName, RdbStreamPendingEntry *pendingEntry);
+    RdbRes (*handleStreamNewConsumer)(RdbParser *p, void *userData, RdbBulk consName, long long activeTime, long long seenTime);
+    RdbRes (*handleStreamConsumerPendingEntry)(RdbParser *p, void *userData, RdbBulk consName, RdbStreamPendingEntry *pendingEntry);
```

Comment:
consumer name doesn't stand for itself, there can be two consumers with the same name inside different groups.
either we give each group and consumer a unique ID somehow, or just pass the group name to all the callbacks that take consumer name.

on the other hand, we don't pass a pointer to the current key, so the same as the user can assume that all the callbakcs about a stream details are about the last key we started, we can also assume that all the callbacks about a consumer, relate to the last group, and all the callbacks about PEL of a consumer are about the last consumer.
in that case passing the consumer name is excessive.

Reply:
Second option. Thanks.